pkg/util/util.go

```diff
@@ -53,104 +53,104 @@ func ParseEndpoint(endpoint string) (string, string, error) {
 	return scheme, addr, nil
 }
 
-func SplitCommasAndRemoveOuterBrackets(input string) []string {
-	if strings.HasPrefix(input, "[") {
-		input = strings.Trim(input, "[]")
-	} else if strings.HasPrefix(input, "{") {
-		input = strings.Trim(input, "{}")
-	}
+func BytesToGiB(volumeSizeGiB int64) int64 {
+	return volumeSizeGiB / GiB
+}
 
-	var leftCurly []int
-	var leftSquare []int
-
-	var slices []string
-	lastComma := 0
-
-	for i, c := range input {
-		if "{" == string(c) {
-			leftCurly = append(leftCurly, i)
-		} else if "[" == string(c) {
-			leftSquare = append(leftSquare, i)
-		} else if "}" == string(c) {
-			if len(leftCurly) > 0 {
-				leftCurly = leftCurly[:len(leftCurly)-1]
-			}
-		} else if "]" == string(c) {
-			if len(leftSquare) > 0 {
-				leftSquare = leftSquare[:len(leftSquare)-1]
-			}
-		} else if "," == string(c) {
-			if len(leftCurly) == 0 && len(leftSquare) == 0 {
-				value := input[lastComma:i]
-				slices = append(slices, value)
-				lastComma = i + 1
-			}
+func GiBToBytes(volumeSizeGiB int64) int64 {
+	return volumeSizeGiB * GiB
+}
+
+func Contains(slice []string, element string) bool {
+	for _, value := range slice {
+		if value == element {
+			return true
 		}
 	}
+	return false
+}
 
-	slices = append(slices, input[lastComma:])
+func EncodeDeletionTag(input string) string {
+	input = strings.ReplaceAll(input, "\"", " @ ")
+	input = strings.ReplaceAll(input, ",", " . ")
+	input = strings.ReplaceAll(input, "[", " - ")
+	input = strings.ReplaceAll(input, "]", " _ ")
+	input = strings.ReplaceAll(input, "{", " + ")
+	input = strings.ReplaceAll(input, "}", " = ")
+	return input
+}
 
-	var values []string
-	for _, val := range slices {
-		if strings.HasPrefix(val, "{") {
-			val = strings.Trim(val, "{}")
-		}
-		values = append(values, val)
-	}
-	return values
+func DecodeDeletionTag(input string) string {
+	input = strings.ReplaceAll(input, " @ ", "\"")
+	input = strings.ReplaceAll(input, " . ", ",")
+	input = strings.ReplaceAll(input, " - ", "[")
+	input = strings.ReplaceAll(input, " _ ", "]")
+	input = strings.ReplaceAll(input, " + ", "{")
+	input = strings.ReplaceAll(input, " = ", "}")
+	return input
 }
 
-func MapValues(input []string) (map[string]*string, error) {
-	valueMap := make(map[string]*string)
-	for _, configOption := range input {
-		splitKeyValue := strings.SplitN(configOption, "=", 2)
-		if len(splitKeyValue) == 2 {
-			if splitKeyValue[1] == "" {
-				valueMap[splitKeyValue[0]] = nil
-			} else {
-				valueMap[splitKeyValue[0]] = &splitKeyValue[1]
-			}
-		} else {
-			return nil, errors.New("Input doesn't contain =")
+func ConvertStringMapToAny(input map[string]string) map[string]any {
```

Comment:
This function (and others) should have some comments. Not sure what to name this function, but it should be more clear that we're converting the map values to their associated types... I'll try to think of a suggestion